idpi/src/operators/time_operators.py

```diff
@@ -0,0 +1,21 @@
+# Third-party
+import numpy as np
+import xarray as xr
+
+
+def time_rate(var: xr.DataArray, dtime: np.timedelta64):
+    coord = var.coords["step"]
+    return (
+        var.isel(step=slice(1, None))
```

Comment:
Note that the value of the resulting rate has to be set to undefined (np.nan) for step=0.